pstress/pstress_log_to_sql_converter.sh

```diff
@@ -0,0 +1,138 @@
+Help() {
+  echo "usage: $BASH_SOURCE --logfile <val>"
+  echo "usage: $BASH_SOURCE --logfile <val> --mysql-client <val> --socket <val>"
+  echo "--logfile: Full path to the pstress log file"
+  echo "--mysql-client: Path to MySQL client"
+  echo "--socket: Path to socket file to connect to running server"
+}
+
+if [ "$#" -eq 0 ]; then
+  Help
+  exit
+fi
+
+ARGUMENT_LIST=(
+    "logfile"
+    "mysql-client"
+    "socket"
+    "user"
+    "help"
+)
+
+# Read arguments
+opts=$(getopt \
+    --longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}" | sed 's/.\{2\}$//')" \
+    --name "$(basename "$0")" \
+    --options "" \
+    -- "$@"
+)
+
+if [ $? -ne 0 ]; then
+  echo "Invalid option. Exiting..."
+  exit 1
+fi
+
+eval set --$opts
+
+while true; do
+    case "$1" in
+    --help)
+        Help
+        exit
+        ;;
+    --mysql-client)
+	shift
+	MYSQL=$1
+	;;
+    --socket)
+	shift
+	SOCKET=$1
+	;;
+    --logfile)
+        shift
+	LOG_FILENAME=$1
+	;;
+    --user)
+	shift
+	USER=$1
+	;;
+    --)
+	Help
+	break
+	;;
+    esac
+    shift
+done
+
+if [ ! -s $LOG_FILENAME ]; then
+  echo "Input File $LOG_FILENAME is empty or does not exist. Exiting..."
+  exit 1
+fi
+
+SCRIPT=$(readlink -f $0)
+SCRIPT_PATH=$(dirname $SCRIPT)
+LOGFILE=$(basename $LOG_FILENAME)
+OUTPUT_FILENAME=$SCRIPT_PATH/reduced_"$LOGFILE"
+
+echo "Reading the pstress logfile: $LOG_FILENAME"
+
+# Filtering out all the successfully executed SQLs from the pstress log
+sed -n 's/.* S //p' $LOG_FILENAME > $OUTPUT_FILENAME
+
+# Trimming off un-necessary information at the beginning of each SQL
+sed -i 's/^ S //g' $OUTPUT_FILENAME
+
+# Trimming off un-necessary information at the end of each SQL
+sed -i 's/rows:[0-9]*//g' $OUTPUT_FILENAME
+
+# Find the crashing query
+crashQuery=$(awk '/Error Lost connection to MySQL server during query/{print prev} {prev=$0}' $LOG_FILENAME | head -n1 | sed 's/.* F //')
+
+# Append the crashing query at the end of output file
+echo $crashQuery >> $OUTPUT_FILENAME
+
+# Adding semi-colon at the end of each SQL statement. In case, semi-colon
+# already exists, then do not add twice
+sed -i '/[^;] *$/s/$/;/' $OUTPUT_FILENAME
+
+echo "Converted SQL file can be found here: $OUTPUT_FILENAME"
+
+#################################################################################################################
+# NOTE:                                                                                                         #
+# 1. Make sure to start a fresh MySQL instance before running this script. Executing the SQL file on an already #
+#    running server may have existing database objects (eg. general tablespaces, tables, etc) causing failures. #
+#################################################################################################################
+# To Execute SQLs against a running server set the MYSQL and SOCKET path (Optional).
+if [[ $MYSQL != "" && $SOCKET != "" ]]; then
+  $MYSQL -u$USER -p -S$SOCKET -e "source $OUTPUT_FILENAME"
```

Comment:
ask for password only if the user option is provided, if the user has not provided use root, in that no need to ask for password.

Reply:
Even for root user, we can set password. It is not always blank

Reply:
okay. but let's connect by default, if the user option is not provided.

Reply:
What's the advantage of doing it?

Reply:
Defaulted the username to root, but we cannot avoid password and it can be set to any value even for root user